internal/reports/report_text.go

```diff
@@ -0,0 +1,271 @@
+		"a table-by-table listing of "+structuredReport.MigrationType+" conversion details. "+
+		"For background on the "+structuredReport.MigrationType+" conversion process used, "+
+		"and explanations of the terms and notes used in this "+
+		"report, see HarbourBridge's README.", 80, 0)
+	w.WriteString("\n\n")
+	if isDump {
+		writeStatementStats(structuredReport, w)
+	}
+	writeNameChanges(structuredReport, w)
+	writeTableReports(structuredReport, w)
+	writeUnexpectedConditionsv2(structuredReport, w)
+
+}
+
+func writeUnexpectedConditionsv2(structuredReport StructuredReport, w *bufio.Writer) {
+	reparseInfo := func() {
+		if structuredReport.UnexpectedConditions.Reparsed > 0 {
+			fmt.Fprintf(w, "Note: there were %d %s reparse events while looking for statement boundaries.\n\n", structuredReport.UnexpectedConditions.Reparsed, structuredReport.StatementStats.DriverName)
+		}
+	}
+	writeHeading(w, "Unexpected Conditions")
+	if len(structuredReport.UnexpectedConditions.UnexpectedConditions) == 0 {
+		w.WriteString("There were no unexpected conditions encountered during processing.\n\n")
+		reparseInfo()
+		return
+	}
+	switch structuredReport.StatementStats.DriverName {
+	case constants.MYSQLDUMP:
+		w.WriteString("For debugging only. This section provides details of unexpected conditions\n")
+		w.WriteString("encountered as we processed the mysqldump data. In particular, the AST node\n")
+		w.WriteString("representation used by the pingcap/tidb/parser library used for parsing\n")
+		w.WriteString("mysqldump output is highly permissive: almost any construct can appear at\n")
+		w.WriteString("any node in the AST tree. The list details all unexpected nodes and\n")
+		w.WriteString("conditions.\n")
+	case constants.PGDUMP:
+		w.WriteString("For debugging only. This section provides details of unexpected conditions\n")
+		w.WriteString("encountered as we processed the pg_dump data. In particular, the AST node\n")
+		w.WriteString("representation used by the pganalyze/pg_query_go library used for parsing\n")
+		w.WriteString("pg_dump output is highly permissive: almost any construct can appear at\n")
+		w.WriteString("any node in the AST tree. The list details all unexpected nodes and\n")
+		w.WriteString("conditions.\n")
+	default:
+		w.WriteString("For debugging only. This section provides details of unexpected conditions\n")
+		w.WriteString("encountered as we processed the " + structuredReport.StatementStats.DriverName + " data. The list details\n")
+		w.WriteString("all unexpected conditions\n")
+	}
+	w.WriteString("  --------------------------------------\n")
+	fmt.Fprintf(w, "  %6s  %s\n", "count", "condition")
+	w.WriteString("  --------------------------------------\n")
+	for _, cond := range structuredReport.UnexpectedConditions.UnexpectedConditions {
+		fmt.Fprintf(w, "  %6d  %s\n", cond.Count, cond.Condition)
+	}
+	w.WriteString("\n")
+	reparseInfo()
+}
+
+func writeTableReports(structuredReport StructuredReport, w *bufio.Writer) {
```

Comment:
it would help to give a sample of the structure as a comment

Reply:
Added.